rusoto/core/src/signature.rs

```diff
@@ -229,6 +229,102 @@ impl SignedRequest {
         self.params = params;
     }
 
+
+    /// http://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-query-string-auth.html
+    pub fn generate_presigned_url(&mut self, creds: &AwsCredentials) -> String {
+        debug!("Presigning request URL");
+
+        self.sign(creds);
+        let hostname = match self.hostname {
+            Some(ref h) => h.to_string(),
+            None => build_hostname(&self.service, &self.region),
+        };
+
+        let current_time = now_utc();
+        let current_time_fmted = current_time.strftime("%Y%m%dT%H%M%SZ").unwrap();
+        let current_time_fmted = format!("{}", &current_time_fmted);
+        let current_date = current_time.strftime("%Y%m%d").unwrap();
+
+        self.remove_header("x-amz-content-sha256");
+        self.remove_header("X-Amz-Content-Sha256");
```

Comment:
Looking at the implementation of [`remove_header()`](https://github.com/rusoto/rusoto/pull/1018/files#diff-16838c06bc1dac9b3c291ec077aaf713R198) the header name is converted to lower case. So, calling `remove_headers()` twice, with different lower/upper casing, should not be necessary.